plugins/inputs/docker_log/docker_logs.go

```diff
@@ -0,0 +1,312 @@
+package docker_log
+
+import (
+	"context"
+	"crypto/tls"
+	"io"
+	"log"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/filter"
+	"github.com/influxdata/telegraf/internal"
+	tlsint "github.com/influxdata/telegraf/internal/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type DockerLogs struct {
+	Endpoint string
+
+	Timeout internal.Duration
+
+	LabelInclude []string `toml:"docker_label_include"`
+	LabelExclude []string `toml:"docker_label_exclude"`
+
+	ContainerInclude []string `toml:"container_name_include"`
+	ContainerExclude []string `toml:"container_name_exclude"`
+
+	ContainerStateInclude []string `toml:"container_state_include"`
+	ContainerStateExclude []string `toml:"container_state_exclude"`
+
+	tlsint.ClientConfig
+
+	newEnvClient func() (Client, error)
+	newClient    func(string, *tls.Config) (Client, error)
+
+	client          Client
+	filtersCreated  bool
+	labelFilter     filter.Filter
+	containerFilter filter.Filter
+	stateFilter     filter.Filter
+	opts            types.ContainerListOptions
+	wg              sync.WaitGroup
+	mu              sync.Mutex
+	containerList   map[string]io.ReadCloser
+}
+
+const (
+	defaultEndpoint = "unix:///var/run/docker.sock"
+	LOG_BYTES_MAX   = 1000
+)
+
+var (
+	containerStates = []string{"created", "restarting", "running", "removing", "paused", "exited", "dead"}
+)
+
+var sampleConfig = `
+  ## Docker Endpoint
+  ##   To use TCP, set endpoint = "tcp://[ip]:[port]"
+  ##   To use environment variables (ie, docker-machine), set endpoint = "ENV"
+  endpoint = "unix:///var/run/docker.sock"
+  ## Containers to include and exclude. Globs accepted.
```

Comment:
nitpick: spacing. Readme looks good, ensure they contain the same, valid info.

Reply:
fixed spacing!